### PR TITLE
Feature/cbi 820 transaction duplicate

### PR DESCRIPTION
**Pull request by karthikeyan-cere**
Updating entity with Unique
API to fetch duplicate entries in the transaction table.

### CODE REVIEW

File: src/modules/block-scanner/block-scanner.controller.ts [hunk 1]
```diff
@@ -66,4 +67,14 @@ export class BlockScannerController {
     const result = await this.blockScannerService.restart(postRestartRequestDto.network);
     return 'Restarted Successfully';
   }
+
+  @Post('duplicate')
+  public duplicate(@Headers() headers, @Body() duplicateRequestDto: DuplicateRequestDto): Promise<any>{
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.controller.ts:72]: `duplicate` does not describe the endpoint sense. For instance, `have-duplicates` can be used. Rename everywhere.

File: src/modules/block-scanner/block-scanner.controller.ts [hunk 2]
```diff
@@ -66,4 +67,14 @@ export class BlockScannerController {
...
+    const isNetwork = this.blockScannerService.networkMap.has(duplicateRequestDto.network);
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.controller.ts:73]: `isNetowrk` -> `networkSupported`

File: src/modules/block-scanner/block-scanner.service.ts [hunk 3]
```diff
@@ -274,6 +274,20 @@ export class BlockScannerService implements BlockScannerServiceInterface {
     return true;
   }
 
+  /**
+   * CHecks for duplicate entries
+   * @param startTime start time
+   * @param network network identifier
+   */
+  public async duplicate(startTime: string, network: string): Promise<any> {
+    this.logger.log(`About to check for duplicate enteries in transactions`);
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.service.ts:283]: Add startTime and Network to the log.

File: src/modules/block-scanner/dto/duplicate.dto.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import {ApiProperty} from '@nestjs/swagger';
+
+export class DuplicateRequestDto {
+  @IsNotEmpty()
+  @ApiProperty()
+  public network: string;
+
+  @IsNotEmpty()
+  @ApiProperty()
+  public startTime: string;
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/dto/duplicate.dto.ts:11]: `startTime` should be optional. Check the Jira ticket, please.

File: src/modules/block-scanner/block-scanner.controller.ts [hunk 5]
```diff
@@ -66,4 +67,14 @@ export class BlockScannerController {
     const result = await this.blockScannerService.restart(postRestartRequestDto.network);
     return 'Restarted Successfully';
   }
+
+  @Post('duplicate')
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.controller.ts:71]: Why do we need POST here? It should be GET.

File: src/modules/block-scanner/block-scanner.controller.ts (hunk 1 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.controller.ts]: updated.

File: src/modules/block-scanner/block-scanner.controller.ts (hunk 2 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.controller.ts]: updated

File: src/modules/block-scanner/block-scanner.service.ts (hunk 3 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts]: updated

File: src/modules/block-scanner/dto/duplicate.dto.ts (hunk 4 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/dto/duplicate.dto.ts:11]: updated.

File: src/modules/block-scanner/block-scanner.controller.ts (hunk 5 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.controller.ts]: updated

File: src/modules/block-scanner/block-scanner.service.ts [hunk 6]
```diff
@@ -270,10 +270,28 @@ export class BlockScannerService implements BlockScannerServiceInterface {
     const {api} = this.networkMap.get(network);
 
     this.processOldBlock(api, network);
-    
+
     return true;
   }
 
+  /**
+   * CHecks for duplicate entries
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.service.ts:278]: Check spelling for CHecks

File: src/modules/block-scanner/block-scanner.controller.ts [hunk 7]
```diff
@@ -66,4 +67,14 @@ export class BlockScannerController {
     const result = await this.blockScannerService.restart(postRestartRequestDto.network);
     return 'Restarted Successfully';
   }
+
+  @Get('have-duplicates/:network/:startTime?')
```
Comment (AndreiNavoichyk) [inline src/modules/block-scanner/block-scanner.controller.ts:71]: Pass startTime with via queryParameter

File: src/modules/block-scanner/block-scanner.service.ts (hunk 6 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts]: updated

File: src/modules/block-scanner/block-scanner.controller.ts (hunk 7 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.controller.ts]: done

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**